vllm_spyre/v1/core/scheduler.py

```diff
@@ -411,3 +411,140 @@ def check_batch_tkv_limit(self, request, tkv, running,
                      self._cache_check_batch_tkv_limit[outer_key][inner_key])
 
         return return_value
+
+
+class ChunkedPrefillSpyreScheduler(ContinuousBatchingSpyreScheduler):
+    """ Support of chunked prefill """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        # We want to keep track of requests for which the prefill is ongoing.
+        # Theoretically, only one request can be prefilled at a time, but we
+        # keep a list to be able to batch prefills in the future.
+        self.prefilling: list[Request] = []
+
+    def update_from_output(self, scheduler_output, model_runner_output):
+        assert isinstance(model_runner_output, CBSpyreModelRunnerOutput), (
+            "Expecting an instance of CBSpyreModelRunnerOutput when doing "
+            "chunked prefill.")
+
+        for req in self.prefilling:
+            # replace num_computed_tokens with the exact number of computed
+            # prompt tokens, the current value might be wrong because of padding
+
+            # TODO agree on interface with CBSpyreModelRunnerOutput
+            # req.num_computed_tokens = \
+            # model_runner_output.req_num_computed_tokens[req.request_id]
+            pass
+
+        # Remove completed prefills
+        self.prefilling = [
+            req for req in self.prefilling
+            if req.num_computed_tokens < req.num_prompt_tokens
+        ]
```

Comment:
Not sure if we have a better suggestion but @tjohnson31415 and I were both briefly confused because `self.prefilling` reads like it's a dequeue like `self.waiting` and `self.running` but it's not.

Maybe having the name just not end with `-ing` would help?

Reply:
what about `ongoing_prefills`?